src/Result.php

```diff
@@ -8,6 +8,12 @@ abstract class Result
     /** @return T1 */
     abstract public function unwrap();
 
+    /**
+    @param T1 $default
```

Comment:
Is this right? The type of `default` can be different to the type of the thing being unwrapped, I think?

Reply:
Your assumption is correct. 
I have set the type for the `$default` to `string`, since that is what our code is expecting.

Reply:
So we don't want to specialise this to a `string`, I think, but now I'm wondering whether I was right to say that the two things should be different.

The two design options here are: `unwrapOr($default: T1): T1` and `unwrapOr($default: T1): T2`.

If we go with the second one, you would be able to say things like `Ok('abc').unwrapOr(999)` and actually that doesn't make much sense, because the surrounding code in that case would expect a `string` and not an `int`.

So I think you were right first time, sorry!

Do you want to rebase that commit out and I'll approve this?